BaseStation/BaseServer/BaseServer.cs

```diff
@@ -37,33 +40,19 @@ private static void ClientConnected(object sender, EventArgs e)
             Console.WriteLine(Scarlet.Communications.Server.GetClients());
         }
 
-        private static short PreventOverflow(short shortVal)
-        {
-            if (shortVal == -32768)
-            {
-                shortVal++;
-            }
-            return shortVal;
-        }
+        private static long lastControlSend = 0;
 
-        public static void EventLoop()
+        public static void Update(Controller controller)
         {
-            while (!shutdown)
+            if (controller.IsConnected && lastControlSend > CONTROL_SEND_INTERVAL)
```

Comment:
The second portion of this statement always appears to be false? Since lastControlSend is 0 and will never enter the loop.